src/cli.c

```diff
@@ -0,0 +1,115 @@
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <limits.h>
+#include <getopt.h>
+#include <stdio.h>
+#include <errno.h>
+
+#include "cli.h"
+#include "logging.h"
+
+/******************************************************
+ *** LOCAL DEFINES                                  ***
+ ******************************************************/
+
+#define MAX_HZ (1000000L)
+
+/******************************************************
+ *** LOCAL VARIABLES                                ***
+ ******************************************************/
+
+static const char* usage_str = "yobemag [-l <0..4>] <ROM>";
+
+/******************************************************
+ *** LOCAL METHODS                                  ***
+ ******************************************************/
+
+static int safe_strtol(const char* const str_to_conv, int* const store_into) {
+    char* end;
+    const long strtol_in = strtol(str_to_conv, &end, 10);
+
+    errno = 0;
+
+    if (end == str_to_conv) {
+        LOG_FATAL("%s: not a decimal number", str_to_conv);
+    } else if ('\0' != *end) {
+        LOG_FATAL("%s: extra characters at end of input: %s", str_to_conv, end);
+    } else if ((LONG_MIN == strtol_in || LONG_MAX == strtol_in) && ERANGE == errno) {
+        LOG_FATAL("%s out of range of type long", str_to_conv);
+    } else if (strtol_in > INT_MAX) {
+        LOG_FATAL("%ld greater than INT_MAX", strtol_in);
+    } else if (strtol_in < INT_MIN) {
+        LOG_FATAL("%ld less than INT_MIN", strtol_in);
+    } else {
+        *store_into = (int)strtol_in;
+        return EXIT_SUCCESS;
+    }
+
+    return EXIT_FAILURE;
+}
+
+/******************************************************
+ *** EXPOSED METHODS                                ***
+ ******************************************************/
+
+CLIArguments* cli_config_default(void) {
+    CLIArguments* conf = malloc(sizeof(CLIArguments));
+
+    conf->logging_level = (int) FATAL;
+    conf->us_delay = 0;
+    conf->rom_path = "";
+
+    return conf;
+}
+
+void cli_config_destroy(CLIArguments *const conf) {
+    free(conf);
+}
+
+int cli_config_handle(CLIArguments* const conf, int const argc, char **const argv) {
```

Comment:
For const I would follow the rule that it goes as far left as possible until it "hits" a *

Reply:
Could you explain this further please

Reply:
Essentially, you could do something like `char const* const s;`, before and after an asterisk the order of types and const doesn't matter (with regards to the grammar). Pushing const to the left _or_ to the right just makes it consistent.

In the quoted line you used "int const argc" which is not consitent with the rest, that's why I mentioned this "rule" of pushing const to the left

Reply:
I see, so `const int argc` would be the more "correct" parameter here?

Reply:
Yes, "conventionwise" at least